### ISSUE TITLE
list can not be created with LYD_PATH_OPT_UPDATE option

### ISSUE THREAD

**Issue by lukasmacko** [CESNET/libyang#49]
`lyd_new_path` requires value argument to be passed with `LYD_PATH_OPT_UPDATE` flags even for list.

```C
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <libyang/libyang.h>

int main(int argc, char **argv)
{
   struct ly_ctx *ctx = ly_ctx_new(".");
   const struct lys_module *module = ly_ctx_load_module(ctx, "test-module", NULL);
   struct lyd_node *node = lyd_new_path(NULL, ctx, "/test-module:list[key='abc']", NULL, LYD_PATH_OPT_UPDATE);
   if (NULL == node) {
        puts("Fail");
   }
   lyd_free_withsiblings(node);
   ly_ctx_destroy(ctx, NULL);
   return 0;
}

```

**Comment by michalvasko** [comment]
Hi,
please try it again with the current master, I think it was fixed some days ago, but after our 0.8 release.

Regards,
Michal

PS: Nice bug reporting!

**Comment by lukasmacko** [comment]
It's working in 0.8.47
